framework/src/modules/nft/cc_commands/cc_transfer.ts

```diff
@@ -104,24 +104,29 @@ export class CrossChainTransferCommand extends BaseCCCommand {
 
 		if (nftChainID.equals(ownChainID)) {
 			const storeData = await nftStore.get(getMethodContext(), nftID);
+
 			if (status === CCM_STATUS_CODE_OK) {
-				storeData.owner = recipientAddress;
 				const storedAttributes = storeData.attributesArray;
 				storeData.attributesArray = this._internalMethod.getNewAttributes(
 					nftID,
 					storedAttributes,
 					receivedAttributes,
 				);
-				await nftStore.save(getMethodContext(), nftID, storeData);
-				await this._internalMethod.createUserEntry(getMethodContext(), recipientAddress, nftID);
-				await escrowStore.del(getMethodContext(), escrowStore.getKey(sendingChainID, nftID));
-			} else {
+			}
+
+			if (status !== CCM_STATUS_CODE_OK) {
```

Comment:
Why replace `if else` with slower `if if` ?

Reply:
1. The `getMethodContext()` function is called multiple times. If this function is computationally expensive, it might be beneficial to call it once and store the result in a variable.

2. The `escrowStore.getKey(sendingChainID, nftID)` is also called twice. This could be stored in a variable to avoid duplicate computation.

Reply:
sure, it'd be nice to also update the other places which can be improved. but was there a specific reason to change from `if else` to `if if` ?

Reply:
Main reason was to remove the duplication of the calls below

```ts
await nftStore.save(getMethodContext(), nftID, storeData);
await this._internalMethod.createUserEntry(getMethodContext(), recipientAddress, nftID);
await escrowStore.del(getMethodContext(), escrowStore.getKey(sendingChainID, nftID));

Reply:
If we optimize this code for performance, 

```ts
if (status === CCM_STATUS_CODE_OK) {
  const { attributesArray: storedAttributes } = storeData;
  storeData.attributesArray = this._internalMethod.getNewAttributes(
    nftID,
    storedAttributes,
    receivedAttributes,
  );
} else {
 recipientAddress = senderAddress;
}

const { attributesArray } = storeData;
const createNFTEntryPromise = this._internalMethod.createNFTEntry(
  getMethodContext(),
  recipientAddress,
  nftID,
  attributesArray,
);
const createUserEntryPromise = this._internalMethod.createUserEntry(
  getMethodContext(),
  recipientAddress,
  nftID,
);
const delPromise = escrowStore.del(
  getMethodContext(),
  escrowStore.getKey(sendingChainID, nftID),
);

await Promise.all([createNFTEntryPromise, createUserEntryPromise, delPromise]);
```

would be a better option as it can significantly reduce the overall execution time of a group of asynchronous operations when compared to executing them sequentially